src/plugins/cache/cache.c

```diff
@@ -331,7 +331,15 @@ int elektraCacheGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key * pa
 	// load cache from storage
 	keySetString (cacheFile, cacheFileName);
 	elektraFree (cacheFileName);
-	if (ch->cacheStorage->kdbGet (ch->cacheStorage, returned, cacheFile) == ELEKTRA_PLUGIN_STATUS_SUCCESS)
+
+	Key * globalCutpoint = keyNew ("system:/elektra/internal", KEY_END);
+	KeySet * globalCut = ksCut (ch->cacheStorage->global, globalCutpoint);
+	int result = ch->cacheStorage->kdbGet (ch->cacheStorage, returned, cacheFile);
+	ksAppend (ch->cacheStorage->global, globalCut);
+	ksDel (globalCut);
+	keyDel (globalCutpoint);
+
+	if (result == ELEKTRA_PLUGIN_STATUS_SUCCESS)
```

Comment:
@mpranj I had a problem with the cache, because I wrote pointers into the global keyset. Of course those can't be cached. I added this work around to exclude `system:/elektra/internal` from the global keyset cache.

Do you think there could be a better solution? Is the caching of the global keyset really needed? It is somewhat specific to each `KDB` instance, so caching doesn't really make sense IMO.

Maybe instead of explicitly excluding keys from the cache, we should only include `system:/elektra/cached`? That might be a bit more newcomer friendly...

Reply:
> Is the caching of the global keyset really needed?

Yes. The whole concept was introduced to make the cache plugin possible and currently it is definitely needed.

> Maybe instead of explicitly excluding keys from the cache, we should only include system:/elektra/cached? That might be a bit more newcomer friendly...

Retrospectively I fully agree with you that this is confusing. I think we should implement your proposal (and afaik this is a quite small change).

Reply:
Okay, I'll change it so that only `system:/elektra/cached` is cached. Should be pretty simple, since only `cache`/`mmapstorage` and the stuff I added in this PR use the global keyset AFAIK.